base/file.jl

```diff
@@ -510,22 +510,47 @@ function mktemp(parent::AbstractString=tempdir(); cleanup::Bool=true)
     return (filename, Base.open(filename, "r+"))
 end
 
+let
+global tempname
+
+struct GUID
```

Comment:
I want to hide the `GUID`  struct definition using a let block, but apparently they escape  let scoping ?  (noted by `@mbauman ` )

Reply:
We don't have local named types so I think you can just get rid of the `let` wrapper.